template/downstream.c.erb

```diff
@@ -35,6 +35,31 @@ void btx_downstream_push_message(
   DL_APPEND(message_iterator_private_data->queue, elt);
 }
 
+// Exposed to the user
+void btx_push_message(void *btx_handle, const bt_message *message) {
+  _btx_push_message((btx_message_iterator_t *)btx_handle, message);
+}
+
+// User by everything
+void btx_downstream_push_message(
```

Comment:
It is called for both, upstream messages and push_messages. Do you think this is the desired behavior? I understood you only wanted it to call `on_push` when `push_messages`.

https://github.com/TApplencourt/metababel/blob/a96b14a311ddcf172d229672ecb3cd053b74cc2d/template/filter.c.erb#L153-L154

Reply:
It's the correct behavior; `on_push` is called for every message sent downstream (either explicitly by callback or implicitly -- like non event message, or message who didn't get user callbacks attached)
Maybe the name is not ideal